game_patch/multi/network.cpp

```diff
@@ -891,6 +891,10 @@ FunHook<void()> multi_stop_hook{
         // Clear server info when leaving
         g_df_server_info.reset();
         multi_stop_hook.call_target();
+        if (rf::local_player) {
+            extern std::unordered_map<rf::Player*, PlayerAdditionalData> g_player_additional_data_map;
```

Comment:
I don't really like this way of accessing this variable. It wasn't intended to be public. 
Could you add in `player.cpp` a function which removes the player from the map? e.g. `remove/reset_player_additional_data(Player*)`